CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java

```diff
@@ -26,25 +42,40 @@
     final static String TWITTER_PACKAGE_NAME = "com.twitter.android";
     final static String FACEBOOK_PACKAGE_NAME = "com.facebook.katana";
     final static int REQUEST_CODE = 100;
-    static public void postToTwitter(String message, String imagePath) {
-        String url = "http://twitter.com/share?text=" + message;
-        Intent intent = new Intent(Intent.ACTION_SEND, Uri.parse(url));
-        intent.putExtra(Intent.EXTRA_TEXT, message);
-        if (!imagePath.equals("")) {
-            try {
-                Uri originalImageUri = Uri.parse(imagePath);
-                String ext = getExtension(originalImageUri);
-                Uri readableFileUri = saveImageToExternalDirectory(originalImageUri);
-                if (readableFileUri != null) {
-                    intent.setType("image/" + ext);
-                    intent.putExtra(Intent.EXTRA_STREAM, readableFileUri);
+    private static TwitterAuthClient authClient = null;
+    private static File imageFile = null;
```

Comment:
I think it should not have variables as variables...

Reply:
Do you say I should use local variable instead of member variable?
authClient member variable need to use `onActivityResult`.
imageFile use only less memory allocation. It can remove but SmartPhone does not have huge memory. It is a little be friendly.